src/App.tsx

```diff
@@ -1,61 +1,163 @@
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+
+function getUser(id: number) {
+  return usersFromServer.find(person => id === person.id);
+}
+
+const todosWithUser = todosFromServer.map(todo => {
+  const user = getUser(todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+function getId(todos: Todo[]) {
+  const ids = todos.map(todo => todo.id);
+  const newId = Math.max(...ids) + 1;
+
+  return newId;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [formValues, setFormValues] = useState({
+    todoTitle: '',
+    todoUser: '0',
+  });
+  const [formErrors, setFormErrors] = useState({
+    titleError: false,
+    userError: false,
+  });
+
+  function formReset() {
+    setFormValues({
+      todoTitle: '',
+      todoUser: '0',
+    });
+  }
+
+  function handleSubmitErrors() {
+    if (!formValues.todoTitle.trim()) {
+      setFormErrors(prevErrors => ({
+        ...prevErrors,
+        titleError: true,
+      }));
+    }
+
+    if (!+formValues.todoUser) {
+      setFormErrors(prevErrors => ({
+        ...prevErrors,
+        userError: true,
+      }));
+    }
+  }
+
+  function handleChanges(event: React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLSelectElement>) {
+    setFormValues(prevFormValues => ({
+      ...prevFormValues,
+      [event.target.name]: event.target.value,
+    }));
+    if (formErrors.titleError && event.target.name === 'todoTitle') {
+      setFormErrors(prevErrors => ({
+        ...prevErrors,
+        titleError: false,
+      }));
+    }
+
+    if (formErrors.userError && event.target.name === 'todoUser') {
+      setFormErrors(prevErrors => ({
+        ...prevErrors,
+        userError: false,
+      }));
+    }
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    const user = getUser(+formValues.todoUser);
+
+    const newTodo = {
+      id: getId(todos),
+      title: formValues.todoTitle,
+      completed: false,
+      userId: +formValues.todoUser,
```

Comment:
additionally to my previous comment, it looks some strange when you assign `todoUser` to `userId`.